packages/zowe-explorer/src/trees/shared/SharedUtils.ts

```diff
@@ -361,6 +361,31 @@ export class SharedUtils {
         };
     }
 
+    /**
+     * Debounces an async event callback to prevent duplicate triggers.
+     * @param callback Async event callback
+     * @param delay Number of milliseconds to delay
+     */
+    public static debounceAsync<T extends (...args: any[]) => Promise<any>>(
```

Comment:
It'd be cool to move this into the Zowe SDKs (same with the regular `debounce`) 😋